src/deterministic_vi/elbo_objective.jl

```diff
@@ -36,39 +36,54 @@ function calculate_source_pixel_brightness!{NumType <: Number}(
     @inbounds for i in 1:Ia # Stars and galaxies
         fsm_i = (i == 1) ? fs0m : fs1m
         a_i = ea.vp[s][ids.a[i, 1]]
+        sb_E_l_a_b_i = sb.E_l_a[b, i]
```

Comment:
Did you try to benchmark the effect of these loads? I tried a few without where I couldn't see an effect and in contrast to most of the other changes which make the code nicer, this makes it quite a bit uglier.